org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelCodeLensParticipant.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lemminx.extensions.contentmodel.participants;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.extensions.contentmodel.commands.AssociateGrammarCommand;
+import org.eclipse.lemminx.extensions.contentmodel.commands.AssociateGrammarCommand.GrammarBindingType;
+import org.eclipse.lemminx.services.extensions.codelens.ICodeLensParticipant;
+import org.eclipse.lemminx.services.extensions.codelens.ICodeLensRequest;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.CodeLens;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+/**
+ * When XML file is not associated to a grammar (XSD, DTD), this class generates
+ * several CodeLenses on the root of the DOM Document:
+ * 
+ * <ul>
+ * <li>[Bind with XSD] : click on this Codelens open a folder dialog to select
+ * the XSD to bind.</li>
+ * <li>[Bind with DTD] : click on this Codelens open a folder dialog to select
+ * the DTD to bind.</li>
+ * <li>[Bind with xml-model] : click on this Codelens open a folder dialog to
+ * select the XSD, DTD to bind.</li>
+ * </ul>

```

Comment:
Add a note that LemMinX will call the `xml.associate.grammar.selectFile` command when the code lens is clicked, which the client needs to implement.

Reply:
Fixed